builtin/providers/aws/cloudfront_distribution_configuration_structure.go

```diff
@@ -375,6 +393,58 @@ func flattenTrustedSigners(ts *cloudfront.TrustedSigners) []interface{} {
 	return []interface{}{}
 }
 
+func expandLambdaFunctionAssociations(v interface{}) *cloudfront.LambdaFunctionAssociations {
+	if v == nil {
+		return &cloudfront.LambdaFunctionAssociations{
+			Quantity: aws.Int64(0),
+		}
+	}
+
+	s := v.(*schema.Set)
+	var lfa cloudfront.LambdaFunctionAssociations
+	lfa.Quantity = aws.Int64(int64(s.Len()))
+	lfa.Items = make([]*cloudfront.LambdaFunctionAssociation, s.Len())
+	for i, lf := range s.List() {
+		lfa.Items[i] = expandLambdaFunctionAssociation(lf.(map[string]interface{}))
+	}
+	return &lfa
+}
+
+func expandLambdaFunctionAssociation(lf map[string]interface{}) *cloudfront.LambdaFunctionAssociation {
+	var lfa cloudfront.LambdaFunctionAssociation
+	if v, ok := lf["event_type"]; ok {
+		lfa.EventType = aws.String(v.(string))
+	}
+	if v, ok := lf["lambda_arn"]; ok {
+		lfa.LambdaFunctionARN = aws.String(v.(string))
+	}
+	return &lfa
+}
+
+func flattenLambdaFunctionAssociations(lfa *cloudfront.LambdaFunctionAssociations) *schema.Set {
+	s := make([]interface{}, len(lfa.Items))
+	for i, v := range lfa.Items {
+		s[i] = flattenLambdaFunctionAssociation(v)
+	}
+	return schema.NewSet(lambdaFunctionAssociationHash, s)
+}
+
+func flattenLambdaFunctionAssociation(lfa *cloudfront.LambdaFunctionAssociation) map[string]interface{} {
+	m := map[string]interface{}{}
+	if lfa != nil {
+		m["event_type"] = *lfa.EventType
+		m["lambda_arn"] = *lfa.LambdaFunctionARN
+	}
+	return m
+}
+
+func lambdaFunctionAssociationHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(m["event_type"].(string))
+	return hashcode.String(buf.String())
```

Comment:
It's actually generally not necessary to provide an explicit set hash function anymore, because `helper/schema` has a default one which uses the schema structure to automatically include everything.

So since this is a new attribute anyway, I'd suggest dropping this function and unsetting the `Set` field on the `Schema` for this attribute. We only really continue to support explicit `Set` for compatibility with schemas that predate the automatic hashing.